### DISCUSSION TITLE
Confused about where to add (simple) customizations

### DISCUSSION THREAD

**Discussion by splooge** [LazyVim/LazyVim#148] [General]
I want to set `update_in_insert` to true, but I'm not sure where to do this.  When I do it in 

```
share/nvim/lazy/LazyVim/lua/lazyvim/plugin/lsp/init.lua
```
it works but that's obviously not the right place to do it since it then conflicts with updating the plugin when using Lazy/sync.

What is the proper place (and format!) to enable this option?  I'm new to lua so If someone could give a complete example on this, that would be appreciated!  I've scoured the docs, and while I found the option set in vim-lspconfig, I'm just to green to make sense of where to place the option to change it without interfering with other things.

Thanks!  And thanks for all the hard work on LazyVim!

**Comment by folke** [comment]
It's literally in the docs https://www.lazyvim.org/plugins/lsp#nvim-lspconfig

Just add the below to one of your specs in `lua/plugins/*.lua`

```lua
  {
    "neovim/nvim-lspconfig",
    opts= {
      diagnostics = {
        update_in_insert = true,
    }
  },
```

**Comment by splooge** [comment]
I still don't get it.  I've tried:

1) adding a file with your above recommendations in it to a (lua/plugin/test.lua), but your recommendations contain 3 open braces and 2 close, which, as a newbie programmer, makes no sense to me.

2) You have 2 "neovim/nvim-lspconfig" settings in example.lua, where I've tried adding your code, and neither work there.  Unfortunately the messages go by too fast for me to get a good read on them.  One message I can see is a "Undefined global `diagnostics`"

3) I've tried copy/pasting both the nvim-lspconfig (option and full spec) from https://github.com/neovim/nvim-lspconfig and still neither of them work for me.

4) Perhaps I'm misunderstanding what "specs" means  Sorry for the newbiness but  I have zero understanding of what you're try to say by "just add this to a spec."

This is why I asked for a working example; because I'm an idiot.

Thanks

**Reply by folke** [reply]
You should probably read one of the lua guides first, since that seems to be your problem.

I did have a typo in there. I can't possibly be expected to fix/check everyone's configs.

But anyway, the code above should be part of  a list, so something like the below:

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts= {
      diagnostics = {
        update_in_insert = true
      }
    }
  }
}
```

Again, you should familiazrize yourself with Lua and Neovim lua specifically, since this is all very basic stuff.
